File: app/src/main/java/bobrchess/of/by/belaruschess/view/activity/RegistrationContractView.java
package bobrchess.of.by.belaruschess.view.activity;

import android.view.View;

import com.arellomobile.mvp.MvpView;

import java.util.List;

import bobrchess.of.by.belaruschess.dto.UserDTO;

/**
 * Created by Igor on 22.04.2018.
 */

public interface RegistrationContractView extends MvpView, BaseContractView, ButtonContractView{
    void registration();
    void showIncorrectEmailText();
    void showIncorrectPasswordText();
    void showNoConnectionAlertDialog(Integer title, Integer message, Integer buttonText, Boolean cancelable);
    void showServerInUnavailableAlertDialog(Integer title, Integer message, Integer buttonText, Boolean cancelable);
    void showSnackBar(View componentView, Integer text, Integer buttonText);
    void showSnackBar(View componentView, String message, Integer buttonText);
    void startActivity(UserDTO userDTO);
    void setGenderSpinnerAdapter(List<String> userNames);
    void setCoachSpinnerAdapter(List<String> userNames);
    void setRankSpinnerAdapter(List<String> rankNames);
    void setCountrySpinnerAdapter(List<String> countryNames);
    void loadSpinnersData();
}

File: app/src/main/java/bobrchess/of/by/belaruschess/view/activity/RegistrationContractView.kt
package bobrchess.of.by.belaruschess.view.activity

import android.view.View

import com.arellomobile.mvp.MvpView

import bobrchess.of.by.belaruschess.dto.UserDTO

/**
 * Created by Igor on 22.04.2018.
 */

interface RegistrationContractView : MvpView, BaseContractView, ButtonContractView {
    fun registration()
    fun showIncorrectEmailText()
    fun showIncorrectPasswordText()
    fun showSnackBar(componentView: View, text: Int)
    fun showSnackBar(componentView: View, message: String)
    fun startActivity(userDTO: UserDTO)
    fun setGenderSpinnerAdapter(userNames: MutableList<String>)
    fun setCoachSpinnerAdapter(userNames: MutableList<String>)
    fun setRankSpinnerAdapter(rankNames: MutableList<String>)
    fun setCountrySpinnerAdapter(countryNames: MutableList<String>)
    fun loadSpinnersData()
}
